### PR TITLE
4296: Review Saved Petition edit buttons

### PR DESCRIPTION
**Pull request by sutt0n**
image

### CODE REVIEW

File: web-client/src/presenter/actions/caseDetailEdit/getCaseCaptionForCaseInfoTabAction.test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { getCaseCaptionForCaseInfoTabAction } from './getCaseCaptionForCaseInfoTabAction';
+import { presenter } from '../../presenter';
+import { runAction } from 'cerebral/test';
+
+presenter.providers.applicationContext = applicationContext;
+
+describe('getCaseCaptionForCaseInfoTabAction', () => {
+  const { Case, ContactFactory } = applicationContext.getEntityConstructors();
+
+  it('should return just the respondent when party type has not been selected', async () => {
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/caseDetailEdit/getCaseCaptionForCaseInfoTabAction.test.js]: This description isn't super clear if you don't know what the contents of `CASE_CAPTION_POSTFIX` are. Maybe `should return just the postfix when party type has not been selected`?

File: web-client/src/presenter/actions/caseDetailEdit/navigateToEditSavedDocumentDetailAction.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { state } from 'cerebral';
+/**
+ * changes the route to view the review
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.router the riot.router object that is used for changing the route
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/caseDetailEdit/navigateToEditSavedDocumentDetailAction.js]: jsdoc updates

File: web-client/src/presenter/actions/caseDetailEdit/navigateToEditSavedDocumentDetailAction.test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        caseDetail: {
+          docketNumber: '101-12',
+        },
+      },
+    });
+
+    expect(routeStub).not.toHaveBeenCalled();
+  });
+
+  it('does not navigate to the edit saved document detail view if document id is not provided', async () => {
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/caseDetailEdit/navigateToEditSavedDocumentDetailAction.test.js]: `if the docket number is not provided`

File: web-client/src/presenter/computeds/formattedCaseDetail.js [hunk 4]
```diff
@@ -108,6 +109,11 @@ export const formattedCaseDetail = (get, applicationContext) => {
           qcWorkItemsUntouched
         ) {
           editLink = '/edit';
+        } else if (
+          document.isPetition &&
+          caseDetail.status === Case.STATUS_TYPES.inProgress
+        ) {
+          editLink = '/edit-saved';
```
Comment (rachaelparris) [inline web-client/src/presenter/computeds/formattedCaseDetail.js:116]: Could you add a test for this new code?

**Review by rachaelparris — COMMENTED** [review]
Just a few minor things

File: web-client/src/presenter/actions/caseDetailEdit/getCaseCaptionForCaseInfoTabAction.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { state } from 'cerebral';
+
+/**
+ * Fetches the case caption using the getCaseCaption helper method
+ *
+ * @param {object} providers the providers object
+ * @param {Function} providers.applicationContext application context to get Case entity
+ * @param {Function} providers.get the cerebral get function
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/caseDetailEdit/getCaseCaptionForCaseInfoTabAction.js:8]: please add `props` to jsdoc

File: web-client/src/presenter/actions/caseDetailEdit/navigateToReviewSavedPetitionAction.test.js [hunk 6]
```diff
@@ -1,13 +1,12 @@
...
 import { presenter } from '../../presenter';
 import { runAction } from 'cerebral/test';
-import sinon from 'sinon';
 
 describe('navigateToReviewSavedPetitionAction', () => {
   let routeStub;
 
   beforeEach(() => {
-    routeStub = sinon.stub();
+    routeStub = jest.fn();
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/caseDetailEdit/navigateToReviewSavedPetitionAction.test.js:9]: 👊

File: web-client/src/presenter/sequences/gotoEditSavedDocumentDetailSequence.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { navigateToEditSavedDocumentDetailAction } from '../actions/caseDetailEdit/navigateToEditSavedDocumentDetailAction';
+import { setCurrentPageAction } from '../actions/setCurrentPageAction';
+import { setDocumentDetailTabAction } from '../actions/setDocumentDetailTabAction';
+
+export const gotoEditSavedDocumentDetailSequence = [
+  setCurrentPageAction('DocumentDetail'),
+  navigateToEditSavedDocumentDetailAction,
```
Comment (kkoskelin) [inline web-client/src/presenter/sequences/gotoEditSavedDocumentDetailSequence.js:7]: Help me understand -- why do we need to set the current page if we are invoking a route?  Shouldn't the route be responsible for setting its own page?  Or is there some nuance I'm missing?

File: web-client/src/views/CaseDetailEdit/ReviewSavedPetition.jsx [hunk 8]
```diff
@@ -116,15 +147,32 @@ export const ReviewSavedPetition = connect(
...
+                    </h3>
                     <div className="grid-row grid-gap">
                       <div className="tablet:grid-col-6 margin-bottom-05">
                         <div className="margin-top-3 margin-bottom-2">
                           <label
                             className="usa-label usa-label-display"
                             htmlFor="filing-type"
                           >
-                            Date recieved
+                            Date received
```
Comment (kkoskelin) [inline web-client/src/views/CaseDetailEdit/ReviewSavedPetition.jsx:175]: 😮

**Review by kkoskelin — CHANGES_REQUESTED** [review]
Just a few observations.  Some top work, sir!

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**